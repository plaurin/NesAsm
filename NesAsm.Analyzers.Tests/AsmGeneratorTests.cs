using ApprovalTests.Reporters;

namespace NesAsm.Analyzers.Tests;

[UseReporter(typeof(VisualStudioReporter))]
public class AsmGeneratorTests
{
    [Fact]
    public void TestExampleScript()
    {
        var generator = new AsmGenerator();
        var content = File.ReadAllText($@"..\..\..\..\NesAsm.Example\HiloWorld.cs");

        GeneratorTestUtilities.TestGenerator(generator, content);
    }

    [Fact]
    public void TestSimpleScript()
    {
        TestScript("SimpleScript.cs");
    }

    [Fact]
    public void TestMultiProcScript()
    {
        TestScript("MultiProcScript.cs");
    }

    [Fact]
    public void TestJumpSubroutineScript()
    {
        TestScript("JumpSubroutineScript.cs");
    }

    [Fact]
    public void TestCharsScript()
    {
        TestScript("CharsScript.cs");
    }

    [Fact]
    public void TestBranchScript()
    {
        TestScript("BranchScript.cs");
    }

    [Fact]
    public void TestParametersScript()
    {
        TestScript("ParametersScript.cs");
    }

    [Fact]
    public void TestSubroutineResultScript()
    {
        TestScript("SubroutineResultScript.cs");
    }

    [Fact]
    public void TestSubroutineResultsScript()
    {
        TestScript("SubroutineResultsScript.cs");
    }

    [Fact]
    public void TestCallingOtherScript()
    {
        TestScript("CallingOtherScript.cs");
    }

    [Fact]
    public void TestDataScript()
    {
        TestScript("DataScript.cs");
    }

    [Fact]
    public void TestAllInstructionsScript()
    {
        TestScript("AllInstructionsScript.cs");
    }

    private static void TestScript(string filename)
    {
        var generator = new AsmGenerator();
        var content = File.ReadAllText($@"..\..\..\TestFiles\{filename}");

        GeneratorTestUtilities.TestGenerator(generator, content);
    }
}
