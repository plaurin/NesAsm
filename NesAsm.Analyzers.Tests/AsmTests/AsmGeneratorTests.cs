using ApprovalTests.Reporters;

namespace NesAsm.Analyzers.Tests.AsmTests;

[UseReporter(typeof(VisualStudioReporter))]
public class AsmGeneratorTests
{
    [Fact]
    public void TestExampleScript()
    {
        var generator = new AsmGenerator();

        GeneratorTestUtilities.TestGenerator(generator, $@"..\..\..\..\NesAsm.Example\HiloWorld\HiloWorld.cs");
    }

    [Fact]
    public void TestSimpleScript()
    {
        TestScript("SimpleScript.cs");
    }

    [Fact]
    public void TestMultiProcScript()
    {
        TestScript("MultiProcScript.cs");
    }

    [Fact]
    public void TestJumpSubroutineScript()
    {
        TestScript("JumpSubroutineScript.cs");
    }

    [Fact]
    public void TestCharsScript()
    {
        TestScript("CharsScript.cs");
    }

    [Fact]
    public void TestBranchScript()
    {
        TestScript("BranchScript.cs");
    }

    [Fact]
    public void TestParametersScript()
    {
        TestScript("ParametersScript.cs");
    }

    [Fact]
    public void TestSubroutineResultScript()
    {
        TestScript("SubroutineResultScript.cs");
    }

    [Fact]
    public void TestSubroutineResultsScript()
    {
        TestScript("SubroutineResultsScript.cs");
    }

    [Fact]
    public void TestCallingOtherScript()
    {
        TestScript("CallingOtherScript.cs", "MultiProcScript.cs");
    }

    [Fact]
    public void TestDataScript()
    {
        TestScript("DataScript.cs");
    }

    [Fact]
    public void TestDataGeneratorScript()
    {
        TestScript("DataGeneratorScript.cs");
    }

    [Fact]
    public void TestAllInstructionsScript()
    {
        TestScript("AllInstructionsScript.cs");
    }

    [Fact]
    public void TestStartupScript()
    {
        TestScript("StartupScript.cs");
    }

    [Fact]
    public void TestIterationStatementsScript()
    {
        TestScript("IterationStatementsScript.cs");
    }

    [Fact]
    public void TestSelectionStatementsScript()
    {
        TestScript("SelectionStatementsScript.cs");
    }

    private static void TestScript(params string[] filenames)
    {
        var generator = new AsmGenerator();

        var sourcePaths = filenames.Select(filename => $@"..\..\..\AsmTests\TestFiles\{filename}").ToArray();

        GeneratorTestUtilities.TestGenerator(generator, sourcePaths);
    }
}
