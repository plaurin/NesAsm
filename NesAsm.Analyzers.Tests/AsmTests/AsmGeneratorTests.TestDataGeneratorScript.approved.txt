- Main diagnostics:
- SyntaxTrees Count: 2
- Output diagnostics:
- RunResult.GeneratedTrees Count: 1
- RunResults diagnostics:
- RunResult[0].Generator: Microsoft.CodeAnalysis.IncrementalGeneratorWrapper
- RunResult[0] diagnostics:
- RunResult[0].GeneratedSources Count: 1
- RunResult[0].Exception: 
- RunResult[0].GeneratedSources[0].HintName: Asm.DataGeneratorScript.cs
- RunResult[0].GeneratedSources[0].SourceText:
/* Generator Asm code in file Asm.DataGeneratorScript.s
; Auto generated code using the NesAsm project
.scope DataGeneratorScript

  .proc start

    ldx #0
  @loop1_on_X:
    lda sprite_data, x
    sta $200, x

    inx
    cpx #20
    bne @loop1_on_X

    rts
  .endproc

.segment "CODE"

sprite_data:
  .byte 48
  .byte 0
  .byte 2
  .byte 47

.segment "CODE"

named_argument:
  .byte 51
  .byte 2
  .byte 1
  .byte 50

.segment "CODE"

mix_ordered_and_named_argument:
  .byte 35
  .byte 8
  .byte 1
  .byte 36

.segment "CODE"

changed_argument_order:
  .byte 62
  .byte 4
  .byte 3
  .byte 63

.segment "CODE"

base_name:
  .byte 75
  .byte 6
  .byte 0
  .byte 74

.endscope


*/
